src/linear_algebra/affine_space/pointwise.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2022 Hanting Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Hanting Zhang
+-/
+import linear_algebra.affine_space.affine_subspace
+
+
+/-! # Pointwise instances on `affine_subspace`s
+
+This file provides:
+
+* `affine_subspace.has_vadd`
+
+and the actions
+
+* `affine_subspace.add_action`
+
+-/
+
+
+noncomputable theory
+open_locale big_operators classical affine pointwise
+
+open set
+
+
+namespace affine_subspace
+variables {k : Type*} {V : Type*} {P : Type*} [ring k] [add_comm_group V] [module k V]
+          [affine_space V P]
+include V
+
+instance : has_vadd V (affine_subspace k P) :=
+{ vadd := λ x S, S.map (affine_equiv.const_vadd k P x) }
+
+instance : add_action V (affine_subspace k P) :=
+{ zero_vadd := λ p,
+    (congr_arg (λ f, p.map f) $ affine_map.ext $ by exact zero_vadd _).trans (p.map_id),
+  add_vadd := λ x y p,
+    (congr_arg (λ f, p.map f) $ affine_map.ext $ by exact add_vadd _ _).trans (p.map_map _ _).symm }
+
+@[simp] lemma coe_const_vadd (v : V) (s : affine_subspace k P) :
```

Comment:
I think this should be called `coe_pointwise_vadd` to match `submodule.coe_pointwise_smul`.

What do you think, @YaelDillies?